README
WordPress-DBIC-Schema

The README is used to introduce the module and provide instructions on
how to install the module, any machine dependencies it may have (for
example C compilers and installed libraries) and any other information
that should be provided before the module is installed.

A README file is required for CPAN modules since CPAN extracts the README
file from a module distribution so that people browsing the archive
can use it to get an idea of the module's uses. It is usually a good idea
to provide version information here so that people can decide whether
fixes for the module are worth downloading.


INSTALLATION

To install this module, run the following commands:

	perl Makefile.PL
	make
	make test
	make install

SUPPORT AND DOCUMENTATION

After installing, you can find documentation for this module with the
perldoc command.

    perldoc WordPress::DBIC::Schema

You can also look for information at:

    RT, CPAN's request tracker (report bugs here)
        https://rt.cpan.org/NoAuth/Bugs.html?Dist=WordPress-DBIC-Schema

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/WordPress-DBIC-Schema

    CPAN Ratings
        https://cpanratings.perl.org/d/WordPress-DBIC-Schema

    Search CPAN
        https://metacpan.org/release/WordPress-DBIC-Schema


LICENSE AND COPYRIGHT

Copyright (C) 2018 Marco Pessotto

This program is free software; you can redistribute it and/or modify it
under the terms of either: the GNU General Public License as published
by the Free Software Foundation; or the Artistic License.

See L<http://dev.perl.org/licenses/> for more information.



Changes
Revision history for WordPress-DBIC-Schema

1.00    2018-7-26

 * First version, with basic relationships



Makefile.PL
use 5.010001;
use strict;
use warnings;
use ExtUtils::MakeMaker;

WriteMakefile(
    NAME             => 'WordPress::DBIC::Schema',
    AUTHOR           => q{Marco Pessotto <melmothx@gmail.com>},
    VERSION_FROM     => 'lib/WordPress/DBIC/Schema.pm',
    ABSTRACT_FROM    => 'lib/WordPress/DBIC/Schema.pm',
    LICENSE          => 'perl_5',
    PL_FILES         => {},
    MIN_PERL_VERSION => '5.010001',
    CONFIGURE_REQUIRES => {
        'ExtUtils::MakeMaker' => '0',
    },
    BUILD_REQUIRES => {
        'Test::More' => '0',
    },
    PREREQ_PM => {
                  'DBIx::Class' => 0,
                  'DBIx::Class::Tree' => 0,
                  'DBIx::Class::Helpers' => 0,
                  'DBIx::Class::Schema::Config' => 0,
                 },
    dist  => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
    clean => { FILES => 'WordPress-DBIC-Schema-*' },
);


lib/WordPress/DBIC/Schema/Result/WpTerm.pm
use utf8;
package WordPress::DBIC::Schema::Result::WpTerm;

# Created by DBIx::Class::Schema::Loader
# DO NOT MODIFY THE FIRST PART OF THIS FILE

=head1 NAME

WordPress::DBIC::Schema::Result::WpTerm

=cut

use strict;
use warnings;

use base 'DBIx::Class::Core';

=head1 COMPONENTS LOADED

=over 4

=item * L<DBIx::Class::InflateColumn::DateTime>

=back

=cut

__PACKAGE__->load_components("InflateColumn::DateTime");

=head1 TABLE: C<wp_terms>

=cut

__PACKAGE__->table("wp_terms");

=head1 ACCESSORS

=head2 term_id

  data_type: 'bigint'
  extra: {unsigned => 1}
  is_auto_increment: 1
  is_nullable: 0

=head2 name

  data_type: 'varchar'
  default_value: (empty string)
  is_nullable: 0
  size: 200

=head2 slug

  data_type: 'varchar'
  default_value: (empty string)
  is_nullable: 0
  size: 200

=head2 term_group

  data_type: 'bigint'
  default_value: 0
  is_nullable: 0

=cut

__PACKAGE__->add_columns(
  "term_id",
  {
    data_type => "bigint",
    extra => { unsigned => 1 },
    is_auto_increment => 1,
    is_nullable => 0,
  },
  "name",
  { data_type => "varchar", default_value => "", is_nullable => 0, size => 200 },
  "slug",
  { data_type => "varchar", default_value => "", is_nullable => 0, size => 200 },
  "term_group",
  { data_type => "bigint", default_value => 0, is_nullable => 0 },
);

=head1 PRIMARY KEY

=over 4

=item * L</term_id>

=back

=cut

__PACKAGE__->set_primary_key("term_id");


# Created by DBIx::Class::Schema::Loader v0.07046 @ 2018-07-15 12:07:23
# DO NOT MODIFY THIS OR ANYTHING ABOVE! md5sum:meW85vK2T7qLX6M+czocGQ


# You can replace this text with custom code or comments, and it will be preserved on regeneration
1;


lib/WordPress/DBIC/Schema/Result/WpPostmeta.pm
use utf8;
package WordPress::DBIC::Schema::Result::WpPostmeta;

# Created by DBIx::Class::Schema::Loader
# DO NOT MODIFY THE FIRST PART OF THIS FILE

=head1 NAME

WordPress::DBIC::Schema::Result::WpPostmeta

=cut

use strict;
use warnings;

use base 'DBIx::Class::Core';

=head1 COMPONENTS LOADED

=over 4

=item * L<DBIx::Class::InflateColumn::DateTime>

=back

=cut

__PACKAGE__->load_components("InflateColumn::DateTime");

=head1 TABLE: C<wp_postmeta>

=cut

__PACKAGE__->table("wp_postmeta");

=head1 ACCESSORS

=head2 meta_id

  data_type: 'bigint'
  extra: {unsigned => 1}
  is_auto_increment: 1
  is_nullable: 0

=head2 post_id

  data_type: 'bigint'
  default_value: 0
  extra: {unsigned => 1}
  is_nullable: 0

=head2 meta_key

  data_type: 'varchar'
  is_nullable: 1
  size: 255

=head2 meta_value

  data_type: 'longtext'
  is_nullable: 1

=cut

__PACKAGE__->add_columns(
  "meta_id",
  {
    data_type => "bigint",
    extra => { unsigned => 1 },
    is_auto_increment => 1,
    is_nullable => 0,
  },
  "post_id",
  {
    data_type => "bigint",
    default_value => 0,
    extra => { unsigned => 1 },
    is_nullable => 0,
  },
  "meta_key",
  { data_type => "varchar", is_nullable => 1, size => 255 },
  "meta_value",
  { data_type => "longtext", is_nullable => 1 },
);

=head1 PRIMARY KEY

=over 4

=item * L</meta_id>

=back

=cut

__PACKAGE__->set_primary_key("meta_id");


# Created by DBIx::Class::Schema::Loader v0.07046 @ 2018-07-15 12:07:23
# DO NOT MODIFY THIS OR ANYTHING ABOVE! md5sum:sawuZCDy56k+RI25R3pp4Q


# You can replace this text with custom code or comments, and it will be preserved on regeneration
1;


lib/WordPress/DBIC/Schema/Result/WpTermTaxonomy.pm
use utf8;
package WordPress::DBIC::Schema::Result::WpTermTaxonomy;

# Created by DBIx::Class::Schema::Loader
# DO NOT MODIFY THE FIRST PART OF THIS FILE

=head1 NAME

WordPress::DBIC::Schema::Result::WpTermTaxonomy

=cut

use strict;
use warnings;

use base 'DBIx::Class::Core';

=head1 COMPONENTS LOADED

=over 4

=item * L<DBIx::Class::InflateColumn::DateTime>

=back

=cut

__PACKAGE__->load_components("InflateColumn::DateTime");

=head1 TABLE: C<wp_term_taxonomy>

=cut

__PACKAGE__->table("wp_term_taxonomy");

=head1 ACCESSORS

=head2 term_taxonomy_id

  data_type: 'bigint'
  extra: {unsigned => 1}
  is_auto_increment: 1
  is_nullable: 0

=head2 term_id

  data_type: 'bigint'
  default_value: 0
  extra: {unsigned => 1}
  is_nullable: 0

=head2 taxonomy

  data_type: 'varchar'
  default_value: (empty string)
  is_nullable: 0
  size: 32

=head2 description

  data_type: 'longtext'
  is_nullable: 0

=head2 parent

  data_type: 'bigint'
  default_value: 0
  extra: {unsigned => 1}
  is_nullable: 0

=head2 count

  data_type: 'bigint'
  default_value: 0
  is_nullable: 0

=cut

__PACKAGE__->add_columns(
  "term_taxonomy_id",
  {
    data_type => "bigint",
    extra => { unsigned => 1 },
    is_auto_increment => 1,
    is_nullable => 0,
  },
  "term_id",
  {
    data_type => "bigint",
    default_value => 0,
    extra => { unsigned => 1 },
    is_nullable => 0,
  },
  "taxonomy",
  { data_type => "varchar", default_value => "", is_nullable => 0, size => 32 },
  "description",
  { data_type => "longtext", is_nullable => 0 },
  "parent",
  {
    data_type => "bigint",
    default_value => 0,
    extra => { unsigned => 1 },
    is_nullable => 0,
  },
  "count",
  { data_type => "bigint", default_value => 0, is_nullable => 0 },
);

=head1 PRIMARY KEY

=over 4

=item * L</term_taxonomy_id>

=back

=cut

__PACKAGE__->set_primary_key("term_taxonomy_id");

=head1 UNIQUE CONSTRAINTS

=head2 C<term_id_taxonomy>

=over 4

=item * L</term_id>

=item * L</taxonomy>

=back

=cut

__PACKAGE__->add_unique_constraint("term_id_taxonomy", ["term_id", "taxonomy"]);


# Created by DBIx::Class::Schema::Loader v0.07046 @ 2018-07-15 12:07:23
# DO NOT MODIFY THIS OR ANYTHING ABOVE! md5sum:Fl/eXgnqldaqu9Ak6ZoEFA


# You can replace this text with custom code or comments, and it will be preserved on regeneration
1;


lib/WordPress/DBIC/Schema/Result/WpTermRelationship.pm
use utf8;
package WordPress::DBIC::Schema::Result::WpTermRelationship;

# Created by DBIx::Class::Schema::Loader
# DO NOT MODIFY THE FIRST PART OF THIS FILE

=head1 NAME

WordPress::DBIC::Schema::Result::WpTermRelationship

=cut

use strict;
use warnings;

use base 'DBIx::Class::Core';

=head1 COMPONENTS LOADED

=over 4

=item * L<DBIx::Class::InflateColumn::DateTime>

=back

=cut

__PACKAGE__->load_components("InflateColumn::DateTime");

=head1 TABLE: C<wp_term_relationships>

=cut

__PACKAGE__->table("wp_term_relationships");

=head1 ACCESSORS

=head2 object_id

  data_type: 'bigint'
  default_value: 0
  extra: {unsigned => 1}
  is_nullable: 0

=head2 term_taxonomy_id

  data_type: 'bigint'
  default_value: 0
  extra: {unsigned => 1}
  is_nullable: 0

=head2 term_order

  data_type: 'integer'
  default_value: 0
  is_nullable: 0

=cut

__PACKAGE__->add_columns(
  "object_id",
  {
    data_type => "bigint",
    default_value => 0,
    extra => { unsigned => 1 },
    is_nullable => 0,
  },
  "term_taxonomy_id",
  {
    data_type => "bigint",
    default_value => 0,
    extra => { unsigned => 1 },
    is_nullable => 0,
  },
  "term_order",
  { data_type => "integer", default_value => 0, is_nullable => 0 },
);

=head1 PRIMARY KEY

=over 4

=item * L</object_id>

=item * L</term_taxonomy_id>

=back

=cut

__PACKAGE__->set_primary_key("object_id", "term_taxonomy_id");


# Created by DBIx::Class::Schema::Loader v0.07046 @ 2018-07-15 12:07:23
# DO NOT MODIFY THIS OR ANYTHING ABOVE! md5sum:q7eDuejLxkqqkDkfFMlaWA


# You can replace this text with custom code or comments, and it will be preserved on regeneration
1;


lib/WordPress/DBIC/Schema.pm
use utf8;
package WordPress::DBIC::Schema;

# Created by DBIx::Class::Schema::Loader
# DO NOT MODIFY THE FIRST PART OF THIS FILE

use strict;
use warnings;

use base 'DBIx::Class::Schema';

__PACKAGE__->load_namespaces;


# Created by DBIx::Class::Schema::Loader v0.07046 @ 2018-07-15 12:06:53
# DO NOT MODIFY THIS OR ANYTHING ABOVE! md5sum:DRt2wnm6VB6Nj529Ii7t8g

=head1 NAME

WordPress::DBIC::Schema - Database schema for WordPress

=head1 DESCRIPTION

This is a basic schema (and mostly a work in progress) for the
WordPress database, which can be used for migrations, data fetching
etc.

It comes with some relationships even if the DB doesn't implement them
with foreign keys.

=head1 SYNOPSIS

  my $schema = WordPress::DBIC::Schema->connect('wordpress');
  foreach my $id (@ARGV) {
      if (my $post = $schema->resultset('WpPost')->published
          ->find($id, {
                       prefetch => ['metas', { wp_term_relationships => { wp_term_taxonomy => 'wp_term' } }],
                      })) {
          foreach my $taxonomy ($post->taxonomies) {
              print $post->post_title . ' ' . $taxonomy->wp_term->name . "\n";
          }
          print $post->clean_url . "\n";
          print $post->permalink . "\n";
          foreach my $meta ($post->metas) {
              if ($meta->meta_key eq 'subtitle') {
                  print $meta->meta_value . "\n";
              }
          }
          print $post->html_body . "\n";
      }
  }


=cut


our $VERSION = '1.00';

__PACKAGE__->load_components(qw/Helper::Schema::QuoteNames
                                Schema::Config
                               /);

=head1 AUTHOR

Marco Pessotto, C<< <melmothx at gmail.com> >>

=head1 BUGS

Please report any bugs or feature requests to the author's email or
just use the CPAN's RT.

=head1 LICENSE

This module is free software and is published under the same terms as
Perl itself.

=cut


1;
